lib/insights_cloud/async/cloud_connector_announce_task.rb

```diff
@@ -0,0 +1,44 @@
+module InsightsCloud
+  module Async
+    class CloudConnectorAnnounceTask < ::Actions::EntryAction
+      def self.subscribe
+        Actions::RemoteExecution::RunHostsJob
+      end
+
+      def self.connector_feature_id
+        @connector_feature_id ||= RemoteExecutionFeature.feature!(ForemanRhCloud::CloudConnector::CLOUD_CONNECTOR_FEATURE).id
+      end
+
+      def plan(job_invocation)
+        return unless connector_playbook_job?(job_invocation)
+
+        plan_self
+      end
+
+      def finalize(*_args)
+        Organization.unscoped.each do |org|
+          presence = ForemanRhCloud::CloudPresence.new(org, logger)
+          presence.announce_to_sources
+        rescue StandardError => ex
+          logger.warn(ex)
+        end
+      end
+
+      def rescue_strategy_for_self
+        Dynflow::Action::Rescue::Skip
+      end
+
+      def connector_playbook_job?(job_invocation)
+        job_invocation&.remote_execution_feature_id == connector_feature_id
```

Comment:
This would not work if someone went and ran the template directly. Is that acceptable? Otherwise we'd have to go through rex features assigned to the job's template.

Reply:
Yes, it's acceptable. If someone is configuring cloud connector manually, we can assume he knows what he's doing and will run the announce rake task later on.